examples/siro_sandbox/sandbox_app.py

```diff
@@ -1022,95 +357,34 @@ def _record_metrics(self, metrics):
-                    + -self.gui_input.mouse_scroll_offset * zoom_sensitivity
-                )
-            else:
-                self.cam_zoom_dist /= (
-                    1.0 + self.gui_input.mouse_scroll_offset * zoom_sensitivity
-                )
-            self.cam_zoom_dist = mn.math.clamp(
-                self.cam_zoom_dist,
-                self._min_zoom_dist,
-                self._max_zoom_dist,
-            )
-
-        # two ways for camera pitch and yaw control for UX comparison:
-        # 1) press/hold ADIK keys
-        self._camera_pitch_and_yaw_wasd_control()
-        # 2) press left mouse button and move mouse
-        self._camera_pitch_and_yaw_mouse_control()
-
-        lookat = self._create_camera_lookat()
-        self.cam_transform = mn.Matrix4.look_at(
-            lookat[0], lookat[1], mn.Vector3(0, 1, 0)
-        )
-
-    def _sim_update_tutorial(self, dt: float):
-        # todo: get rid of this
-        # Keyframes are saved by RearrangeSim when stepping the environment.
-        # Because the environment is not stepped in the tutorial, we need to save keyframes manually for replay rendering to work.
-        self.get_sim().gfx_replay_manager.save_keyframe()
-
-        self._tutorial.update(dt)
-
-        if self.gui_input.get_key_down(GuiInput.KeyNS.SPACE):
-            self._tutorial.skip_stage()
-
-        if self._tutorial.is_completed():
-            self._tutorial.stop_animations()
-            self._sandbox_state = SandboxState.CONTROLLING_AGENT
-        else:
-            self.cam_transform = self._tutorial.get_look_at_matrix()
+    def _set_cursor_style(self, cursor_style):
+        self._pending_cursor_style = cursor_style
 
     def sim_update(self, dt):
-        # todo: pipe end_play somewhere
         post_sim_update_dict: Dict[str, Any] = {}
 
-        if self.gui_input.get_key_down(GuiInput.KeyNS.ESC):
-            self._end_episode()
-            post_sim_update_dict["application_exit"] = True
-
-        if self.gui_input.get_key_down(GuiInput.KeyNS.M):
-            self._end_episode(do_reset=True)
-
         # _viz_anim_fraction goes from 0 to 1 over time and then resets to 0
         viz_anim_speed = 2.0
         self._viz_anim_fraction = (
             self._viz_anim_fraction + dt * viz_anim_speed
         ) % 1.0
 
-        if self._env_episode_active():
-            self._update_task()
-            self._update_grasping_and_set_act_hints()
-
         # Navmesh visualization only works in the debug third-person view
         # (--debug-third-person-width), not the main sandbox viewport. Navmesh
         # visualization is only implemented for simulator-rendering, not replay-
         # rendering.
-        if self.gui_input.get_key_down(GuiInput.KeyNS.N):
-            self.habitat_env._sim.navmesh_visualization = (  # type: ignore
-                not self.habitat_env._sim.navmesh_visualization  # type: ignore
+        if self._sandbox_service.gui_input.get_key_down(GuiInput.KeyNS.N):
```

Comment:
How about directly accessing (defining attribute in __init__ if needed) `gui_input` and `sim` from within `SandboxDriver`?